pkg/infrastructure/attribute_manager.go

```diff
@@ -32,14 +34,37 @@ const ValidationPath string = "/tmp/taxonomy/infraattributes.json#/definitions/I
 type AttributeManager struct {
 	Log            zerolog.Logger
 	Infrastructure infraattributes.Infrastructure
+	Mux            *sync.RWMutex
 }
 
 func NewAttributeManager() (*AttributeManager, error) {
 	content, err := readInfrastructure()
 	if err != nil {
 		return nil, err
 	}
-	return &AttributeManager{Log: logging.LogInit(logging.CONTROLLER, "FybrikApplication"), Infrastructure: content}, nil
+	return &AttributeManager{
+		Log:            logging.LogInit(logging.CONTROLLER, "FybrikApplication"),
+		Infrastructure: content,
+		Mux:            &sync.RWMutex{},
+	}, nil
+}
+
+func (m *AttributeManager) OnError(err error) {
+	m.Log.Error().Err(err).Msg("Error reading infrastructure attributes")
+}
+
+func (m *AttributeManager) GetOptions() monitor.FileMonitorOptions {
+	return monitor.FileMonitorOptions{Path: RegoPolicyDirectory, Extension: ".json"}
+}
+
+func (m *AttributeManager) OnNotify() {
+	content, err := readInfrastructure()
+	if err != nil {
+		m.OnError(err)
+	}
+	m.Mux.Lock()
+	m.Infrastructure = content
+	m.Mux.Unlock()
 }
 
 func readInfrastructure() (infraattributes.Infrastructure, error) {
```

Comment:
If I understand correctly, you are reading the infrastructure because the monitor identified a change ... and you are reading it in order to validate that the data structure is (still) valid.  Perhaps rename the function to validateInfrastructure?  and please add documentation about what it function does.

Reply:
Both reading the new information and storing in-memory and validation.
Added documentation.